### PR TITLE
Add filter by predicate

### PR DESCRIPTION
**Pull request by kibertoad**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -153,6 +153,54 @@ const result1 = awilixManager.getWithTags(diContainer, ['queue'])
...
+)
+
+const awilixManager = new AwilixManager({
+  diContainer,
+  asyncInit: true,
+  asyncDispose: true,
+})
+
+// This will return dependency1
+const result1 = awilixManager.getByPredicate((entry) => entry instanceof QueueConsumerHighPriorityClass)
```
Comment (CarlosGamero) [inline README.md:197]: 🟢 `result1` will be a `Record<String, any>`, right? is the doc accurate with that?

File: lib/awilixManager.ts [hunk 2]
```diff
@@ -132,6 +131,25 @@ export function getWithTags(diContainer: AwilixContainer, tags: string[]): Recor
...
+    return entry[1].enabled !== false
+  })
+
+  const resolvedComponents: Record<string, any> = {}
+  for (const entry of enabledDependencies) {
+    const resolvedElement = diContainer.resolve(entry[0])
+    if (predicate(resolvedElement)) {
+      resolvedComponents[entry[0]] = resolvedElement
+    }
+  }
```
Comment (CarlosGamero) [inline lib/awilixManager.ts]: 🟢 Nit, feel free to ignore.

`entry[0]` I suppose is the dependency key, would the following be a bit more readable?
```
for (const [key] of enabledDependencies) {
...
}
```

**Review by CarlosGamero — APPROVED** [review]
🚀

File: lib/awilixManager.ts (hunk 2 above)
Comment (kibertoad) [reply lib/awilixManager.ts]: great idea, addressed!

File: README.md (hunk 1 above)
Comment (kibertoad) [reply README.md:197]: improved

**Pull request merged**